### PR TITLE
Fastapi + Gunicorn example in README

### PR DESCRIPTION
**Pull request by ashirviskas**
Also provided additional multiprocessing instructions for FastAPI + Gunicorn setup with code examples as per this issue: #810

Signed-off-by: Matas Minelga <minematas@gmail.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -404,6 +404,56 @@ uwsgi --http 127.0.0.1:8000 --wsgi-file myapp.py --callable app
...
+from fastapi import FastAPI
+from prometheus_client import make_asgi_app
+
+app = FastAPI(debug=False)
+
+# Using multiprocess collector for registry
+def make_metrics_app():
+    registry = CollectorRegistry()
+    multiprocess.MultiProcessCollector(registry)
+    return prometheus_client.make_asgi_app(registry=registry)
```
Comment (hkiang01) [inline README.md:437]: @ashirviskas should be based on the [import](https://github.com/prometheus/client_python/pull/812/commits/60ced23a24abbd2694e405a128db37afa8bfd52d#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R429)
```python
return make_asgi_app(registry=registry)
```

Comment (ashirviskas) [reply README.md]: I might be late, but I just fixed it

File: README.md [hunk 2]
```diff
@@ -404,6 +404,56 @@ uwsgi --http 127.0.0.1:8000 --wsgi-file myapp.py --callable app
...
+# Using multiprocess collector for registry
+def make_metrics_app():
+    registry = CollectorRegistry()
+    multiprocess.MultiProcessCollector(registry)
+    return make_asgi_app(registry=registry)
+
+
+metrics_app = make_metrics_app()
+app.mount("/metrics", metrics_app))
+
```
Comment (csmarchbanks) [inline README.md:442]: Nit: extra newline here.

**Review by csmarchbanks — COMMENTED** [review]
Generally looks good to me with a small nit. In addition, the second commit needs to be signed off.

File: README.md [hunk 3]
```diff
@@ -404,6 +404,56 @@ uwsgi --http 127.0.0.1:8000 --wsgi-file myapp.py --callable app
...
+# Using multiprocess collector for registry
+def make_metrics_app():
+    registry = CollectorRegistry()
+    multiprocess.MultiProcessCollector(registry)
+    return make_asgi_app(registry=registry)
+
+
+metrics_app = make_metrics_app()
+app.mount("/metrics", metrics_app))
```
Comment (JoshChristie) [inline README.md:441]: remove double `) at the end of line

**Comment by gal-leib** [comment]
I know this isn't merged yet, but is it good practice?
Since in the documentation it says the Registry should be created in the state of a request, but here it will be created at the startup of the process no?

**Comment by OliverFarren** [comment]
It's the 11 month anniversary of this PR, are we still deliberating or could this be merged in? 

As per the motive of the original #810, having this in the README.md would have saved significant time.

**Comment by csmarchbanks** [comment]
I cannot merge this without the DCO being signed for all commits.

> I know this isn't merged yet, but is it good practice?
Since in the documentation it says the Registry should be created in the context of a request, but here it will be created at the startup of the process no?

As the readme mentions that is best practice to avoid metrics registering themselves with the multiprocess registry. Since the registry is only created inside of the create app function I think this approach is correct.

**Comment by ashirviskas** [comment]
Whats the easiest way I could resign all the commits?

**Review by csmarchbanks — APPROVED** [review]
Looks like it ran when you force pushed (as it should). Thanks!

**Comment by OliverFarren** [comment]
Wow! Thanks everyone for sorting today and for creating the additional PR in the first place.

**Pull request merged**